xbmc/Util.h

```diff
@@ -227,6 +227,12 @@ class CUtil
                                    const std::string& file,
                                    const std::string& destname = "");
 
+  /*!
+   * \brief Delete a file with/without confirmation.
+   *        Depends on confirmation flag in settings.
+   */
+  static bool FileDelWithConfirm(CFileItemPtr pItem);
```

Comment:
1) We are not short on letters. Please change function name to `FileDeleteWithConfirm`
2) Change param type to `const CFileItem&`, please. No need for a (smart) pointer here. Reference is fine.

Reply:
1. Can't be implemented in file utils as there is no access to the settings/flags there.
2-4. Ok

Reply:
> Can't be implemented in file utils as there is no access to the settings/flags there.

why can’t you just include the missing headers?

Reply:
It's not missing headers: "error C2027: use of undefined type 'CGUIComponent'"

Reply:
That's exactly telling that there is an include missing! You need to add

```
#include "guilib/GUIComponent.h"
```

in CFileUtils.h